### ISSUE TITLE
Translation library

### ISSUE THREAD

**Issue by bmnnnkls** [oblique-bit/oblique#9]
I was wondering if you are aware of the discussion on https://github.com/ngx-translate/core/issues/783. ngx-translate seems to be somehow deprecate, a good alternative imo would be https://ngneat.github.io/transloco/

it adds usefull features like lazy loading, scoped translations for libraries, tools for extracting keys and a schematic for migration from ngx-translate and others.

Are there any plans for the future of oblique regarding translations?

**Comment by gillerr** [comment]
Yes we are fully aware of the state of ngx-translate. But replacing it is not a top priority, mainly because:
- it is maintained (up to date with Angular 10)
- there's no real alternatives (at least yet)
- it will be a huge breaking change, not for Oblique itself, but it will impact all the projects that uses it.

Now the cost of managing the breaking changes could be mitigated with the schematics. Transloco, for instance, even offer its own tool to ease the migration. But it's still a big issue, and we don't want our users to stick to an old version because of this change.

That being said, it seem highly unlikely that @ngx-translate will get significant contribution it the future, meaning we'll have to replace it at some point, not too far away. We'll explore some replacement options next year, and once a tool suitable for our needs will be found, we'll start integrating it. We'll just have to make sure the migration is as little disruptive as possible to avoid our users to skip the update.

**Alternatives**
The alternatives that we're aware of are:
- Angular official plugin
- transloco (https://ngneat.github.io/transloco/)
- Angular I18n (https://github.com/robisim74/angular-l10n)
- locl (https://github.com/loclapp/locl)

Transloco and Angular I18n do have the same inherent problem than @ngx-translate, which is they're all side projects. Progress on the product is totally dependent on the availability of one or a few persons. 

Transloco is very promising but also very young, barely 1 year old and there's not really a community yet. We're not sure it is mature enough, but we sure do keep an eye on it.

Angular I18n has to opposite problem, it is around for as long as @ngx-translate, but never got any traction. And we doubt this will change.

We had great hopes for the official Angular plugin, especially with Oliver Combe working on it. But it doesn't seem to be usable for us in the foreseeable future. There's not even a mention of it in the official Angular Roadmap, which gives a clue about its priority. We really doubt a solution without run time language switching will be applicable in Switzerland.

Locl seems also promising but is not yet production ready. Its licensing also needs to be clarified.

**TL:DR**
- We monitor the situation.
- We'll explore some replacement options next year
- We'll eventually move away from @ngx-translate

**Comment by gillerr** [comment]
Closing due to inactivity
